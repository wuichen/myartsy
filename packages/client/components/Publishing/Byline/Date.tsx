import { Sans, SansSize } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { getDate } from "../Constants"
import { BylineLayout, DateFormat } from "../Typings"

interface DateProps {
  date?: string
  format?: DateFormat
  layout?: BylineLayout
  size?: SansSize
}

export const Date: React.SFC<DateProps> = props => {
  const { date, format, layout, size } = props
  const condensed = layout === "condensed"
  const dateFormat = format || (condensed ? "condensed" : "default")
  const fontSize = size ? size : condensed ? "2" : "3t"

  return (
    <DateContainer condensed={condensed}>
      <Sans size={fontSize} weight="medium">
        {getDate(date, dateFormat)}
      </Sans>
    </DateContainer>
  )
}

export const DateContainer = styled.div.attrs<{ condensed?: boolean }>({})`
  margin: 5px 20px 0 0;
  white-space: nowrap;
`
