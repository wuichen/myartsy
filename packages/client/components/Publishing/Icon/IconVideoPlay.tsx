import React from "react"
import styled from "styled-components"

interface Props extends React.HTMLProps<HTMLDivElement> {
  color?: string
}

const Icon: React.SFC<Props> = props => (
  <svg
    x="0px"
    y="0px"
    viewBox="0 0 40 56"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    className={props.className}
  >
    <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
      <polygon fill={props.color} points="0 0 0 56 39.3600006 28" />
    </g>
  </svg>
)

Icon.defaultProps = {
  color: "black",
}

export const IconVideoPlay = styled(Icon).attrs<{
  suppressClassNameWarning?: boolean
}>({
  suppressClassNameWarning: true,
})`
  width: 32px;
  height: 32px;
`
