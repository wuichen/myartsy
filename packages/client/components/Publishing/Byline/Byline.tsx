import { SansSize } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { getArticleFullHref } from "../Constants"
import { ArticleData, BylineLayout } from "../Typings"
import { Author } from "./Author"
import { Date } from "./Date"
import { Share } from "./Share"

interface BylineProps {
  article: ArticleData
  color?: string
  date?: string
  layout?: BylineLayout
  size?: SansSize
}

export const Byline: React.SFC<BylineProps> = props => {
  const {
    article: { authors, published_at, slug, social_title, thumbnail_title },
    color,
    date,
    layout,
    size,
  } = props

  return (
    <BylineContainer className="Byline" color={color}>
      <Author authors={authors} color={color} layout={layout} size={size} />
      <Date date={date || published_at} layout={layout} size={size} />

      {layout !== "condensed" && (
        <Share
          url={getArticleFullHref(slug)}
          title={social_title || thumbnail_title}
          color={color}
        />
      )}
    </BylineContainer>
  )
}

Byline.defaultProps = {
  color: "black",
}

export const BylineContainer = styled.div.attrs<{ color: string }>({})`
  display: flex;
  flex-wrap: wrap;
  align-items: flex-end;
  color: ${props => props.color};
`
