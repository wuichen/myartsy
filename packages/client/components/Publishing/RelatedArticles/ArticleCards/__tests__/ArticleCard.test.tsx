import { Date } from "Components/Publishing/Byline/Date"
import {
  SeriesArticle,
  StandardArticle,
  VideoArticle,
} from "Components/Publishing/Fixtures/Articles"
import { EditableChild } from "Components/Publishing/Fixtures/Helpers"
import { IconVideoPlay } from "Components/Publishing/Icon/IconVideoPlay"
import { mount } from "enzyme"
import "jest-styled-components"
import { cloneDeep, extend } from "lodash"
import React from "react"
import renderer from "react-test-renderer"
import { ArticleCard, ArticleCardContainer } from "../ArticleCard"

describe("ArticleCard", () => {
  let videoArticle
  let props
  let editProps

  beforeEach(() => {
    videoArticle = cloneDeep(VideoArticle)

    props = {
      article: cloneDeep(StandardArticle),
      series: cloneDeep(SeriesArticle),
      tracking: {
        trackEvent: jest.fn(),
      } as any,
    }

    editProps = extend(cloneDeep(props), {
      editDate: EditableChild("date"),
      editDescription: EditableChild("description"),
      editImage: EditableChild("image"),
      editTitle: EditableChild("title"),
    })
  })

  const getWrapper = (passedProps = props) => {
    return mount(<ArticleCard {...passedProps} />)
  }

  it("renders an article properly", () => {
    const component = renderer.create(<ArticleCard {...props} />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with unpublished media properly", () => {
    props.article = videoArticle
    const component = renderer.create(<ArticleCard {...props} />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("renders an article with children properly", () => {
    editProps.article = videoArticle
    const component = renderer.create(<ArticleCard {...editProps} />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("Renders media duration and play icon if article has media and is published", () => {
    props.article = videoArticle
    props.article.media.published = true
    const component = getWrapper()

    expect(component.find(IconVideoPlay).length).toBe(1)
    expect(component.text()).toMatch("02:28")
  })

  it("Renders coming soon and available date if article has media and is unpublished", () => {
    props.article = videoArticle
    props.article.media.published = false
    const component = getWrapper()

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch("03:12")
    expect(component.text()).toMatch("Coming Soon")
    expect(component.text()).toMatch("Available")
  })

  it("Renders publish date if layout does not have media", () => {
    const component = getWrapper()

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch("Coming Soon")
    expect(component.text()).toMatch("May 19, 2017")
  })

  it("Renders publish date if layout does not have media values", () => {
    const component = getWrapper()

    expect(component.find(IconVideoPlay).length).toBe(0)
    expect(component.text()).not.toMatch("Coming Soon")
    expect(component.text()).toMatch("May 19, 2017")
  })

  it("Does not render byline if article has media", () => {
    props.article = videoArticle
    const component = getWrapper()

    expect(component.find(".author").length).toBe(0)
  })

  it("Renders short media release_date if present and media is unpublished", () => {
    props.article = videoArticle
    props.article.media.published = false
    const component = getWrapper()

    expect(component.text()).toMatch("AvailableAug 2017")
  })

  it("Renders the media release_date if present and media is published", () => {
    props.article = videoArticle
    const component = getWrapper()
    const renderedDate = component
      .find(Date)
      .at(0)
      .text()

    expect(renderedDate).toBe("Aug 28, 2017")
  })

  it("Renders short article.published_at date if no release_date and media is present and unpublished", () => {
    props.article = videoArticle
    props.article.media.published = false
    delete props.article.media.release_date
    const component = getWrapper()

    expect(component.text()).toMatch("AvailableJul 2017")
  })

  it("Renders article.published_at date if no release_date and media is present and published", () => {
    props.article = videoArticle
    props.article.media.published = true
    delete props.article.media.release_date
    const component = getWrapper()
    const renderedDate = component
      .find(Date)
      .at(0)
      .text()

    expect(renderedDate).toBe("Jul 28, 2017")
  })

  it("Renders editable fields if present", () => {
    props = editProps
    props.article = videoArticle
    const component = getWrapper()

    expect(component.text()).toMatch("Child date")
    expect(component.text()).toMatch("Child description")
    expect(component.text()).toMatch("Child image")
    expect(component.text()).toMatch("Child title")
  })

  describe("Analytics", () => {
    it("tracks article card click", () => {
      const component = getWrapper()
      component
        .find(ArticleCardContainer)
        .first()
        .simulate("click")

      expect(props.tracking.trackEvent).toBeCalledWith({
        action: "Click",
        label: "Related article card",
      })
    })
  })
})
