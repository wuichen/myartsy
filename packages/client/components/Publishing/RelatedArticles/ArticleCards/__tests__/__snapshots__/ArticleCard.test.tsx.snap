// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders an article properly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  padding: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0px;
  width: 100%;
}

.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 4px;
}

.c4 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 20px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  margin-bottom: 20px;
}

.c7 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c9 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c8 {
  margin: 10px 20px 0 0;
}

.c8::before {
  content: "";
  display: inline-block;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 0 5px 1px 0;
  background-color: #000;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: #000;
}

.c12 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c10 {
  margin-bottom: 10px;
  margin-left: 0px;
  width: 100%;
  position: relative;
  height: inherit;
  background: white;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: #000;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.c0 .c11 {
  opacity: 1;
}

.c0:hover .c11 {
  opacity: 0.7;
}

@media screen and (min-width:40em) {
  .c1 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    padding: 20px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 28px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    line-height: 36px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 18px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    line-height: 26px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c10 {
    margin-bottom: 10px;
    margin-left: 0px;
  }
}

@media screen and (min-width:52em) {
  .c10 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:64em) {
  .c10 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:40em) {
  .c10 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c10 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c10 {
    width: 50%;
  }
}

<a
  className="ArticleCard c0"
  color="#000"
  href="new-yorks-next-art-district"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <div>
        <div
          className="c3"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          The Future of Art
        </div>
        <div
          className="c4"
          fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
          fontSize={
            Array [
              "28px",
              "28px",
              "42px",
              "42px",
            ]
          }
        >
          New York's Next Art District
        </div>
        <div
          className="c5"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "18px",
              "22px",
              "22px",
            ]
          }
        >
          Land exhibitions, make influential contacts, and gain valuable feedback about your work.
        </div>
      </div>
      <div
        className="Byline c6"
        color="#000"
      >
        <div
          className="c7"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          <div
            className="c8"
            color="#000"
          >
            Casey Lesser
          </div>
        </div>
        <div
          className="c9"
        >
          <div
            className="c7"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize="14px"
          >
            May 19, 2017
          </div>
        </div>
      </div>
    </div>
    <div
      className="c10"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <img
        className="c11 c12"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F7lsxxsw0qPAuKl37jEYitw%252Farticle%2Basset%2B1-hig%2Bres%2Bcopy.jpg&width=680&height=450&quality=95"
      />
    </div>
  </div>
</a>
`;

exports[`ArticleCard renders an article with children properly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  padding: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0px;
  width: 100%;
}

.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 4px;
}

.c4 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 20px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  margin-bottom: 20px;
}

.c6 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c10 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1em;
}

.c9 {
  width: 32px;
  height: 32px;
}

.c7 {
  margin-bottom: 10px;
  margin-left: 0px;
  width: 100%;
  position: relative;
  height: inherit;
  background: white;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: #000;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.c0 .c11 {
  opacity: 1;
}

.c0:hover .c11 {
  opacity: 0.7;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: absolute;
  left: 20px;
  right: 20px;
  bottom: 20px;
}

.c8 svg {
  height: 40px;
}

@media screen and (min-width:40em) {
  .c1 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    padding: 20px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 28px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    line-height: 36px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 18px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    line-height: 26px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c7 {
    margin-bottom: 10px;
    margin-left: 0px;
  }
}

@media screen and (min-width:52em) {
  .c7 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:64em) {
  .c7 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:40em) {
  .c7 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c7 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c7 {
    width: 50%;
  }
}

@media (max-width:900px) {
  .c8 svg {
    height: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="#000"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <div>
        <div
          className="c3"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          The Future of Art
        </div>
        <div
          className="c4"
          fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
          fontSize={
            Array [
              "28px",
              "28px",
              "42px",
              "42px",
            ]
          }
        >
          <div>
            Child 
            title
          </div>
        </div>
        <div
          className="c5"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "18px",
              "22px",
              "22px",
            ]
          }
        >
          <div>
            Child 
            description
          </div>
        </div>
      </div>
      <div
        className="c6"
        fontFamily={
          Object {
            "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
            "fontWeight": 500,
          }
        }
        fontSize="14px"
      >
        <div>
          Child 
          date
        </div>
      </div>
    </div>
    <div
      className="c7"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <div>
        Child 
        image
      </div>
      <div
        className="c8"
      >
        <svg
          className="c9"
          version="1.1"
          viewBox="0 0 40 56"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <polygon
              fill="#000"
              points="0 0 0 56 39.3600006 28"
            />
          </g>
        </svg>
        <div
          className="c10"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          02:28
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`ArticleCard renders an article with unpublished media properly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  padding: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 0px;
  width: 100%;
}

.c3 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  margin-bottom: 4px;
}

.c4 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 28px;
  line-height: 36px;
  margin-bottom: 20px;
}

.c5 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 18px;
  line-height: 26px;
  margin-bottom: 20px;
}

.c7 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c13 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 1em;
}

.c6 {
  margin: 5px 20px 0 0;
  white-space: nowrap;
}

.c12 {
  width: 32px;
  height: 32px;
}

.c10 {
  width: 100%;
  height: 100%;
  object-fit: cover;
  object-position: center;
  -webkit-transition: opacity 0.15s;
  transition: opacity 0.15s;
  display: block;
}

.c8 {
  margin-bottom: 10px;
  margin-left: 0px;
  width: 100%;
  position: relative;
  height: inherit;
  background: white;
}

.c0 {
  border: 1px solid;
  border-radius: 2px;
  color: #000;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
}

.c0 .c9 {
  opacity: 1;
}

.c0:hover .c9 {
  opacity: 0.7;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: absolute;
  left: 20px;
  right: 20px;
  bottom: 20px;
}

.c11 svg {
  height: 40px;
}

@media screen and (min-width:40em) {
  .c1 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    padding: 20px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    padding: 30px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    margin-bottom: 5px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c2 {
    width: 50%;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 28px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    font-size: 42px;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    line-height: 36px;
  }
}

@media screen and (min-width:52em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:64em) {
  .c4 {
    line-height: 50px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    font-size: 18px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    font-size: 22px;
  }
}

@media screen and (min-width:40em) {
  .c5 {
    line-height: 26px;
  }
}

@media screen and (min-width:52em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:64em) {
  .c5 {
    line-height: 32px;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    margin-bottom: 10px;
    margin-left: 0px;
  }
}

@media screen and (min-width:52em) {
  .c8 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:64em) {
  .c8 {
    margin-bottom: 0px;
    margin-left: 30px;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    width: 100%;
  }
}

@media screen and (min-width:52em) {
  .c8 {
    width: 50%;
  }
}

@media screen and (min-width:64em) {
  .c8 {
    width: 50%;
  }
}

@media (max-width:900px) {
  .c11 svg {
    height: 30px;
  }
}

<a
  className="ArticleCard c0"
  color="#000"
  href="joanne-artman-gallery-poetry-naturerefinement-form"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <div
      className="c2"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <div>
        <div
          className="c3"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          The Future of Art
        </div>
        <div
          className="c4"
          fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
          fontSize={
            Array [
              "28px",
              "28px",
              "42px",
              "42px",
            ]
          }
        >
          New Study Shows the Gender Pay Gap for Artists Is Not So Simple
        </div>
        <div
          className="c5"
          fontFamily="'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', Times, serif"
          fontSize={
            Array [
              "18px",
              "18px",
              "22px",
              "22px",
            ]
          }
        >
          The elegant spiral of the Nautilus shell, the sinuous pattern of the banks of a river, or the swirling vortex street of clouds - patterns exist.
        </div>
      </div>
      <div
        className="c6"
      >
        <div
          className="c7"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          Aug 28, 2017
        </div>
      </div>
    </div>
    <div
      className="c8"
      width={
        Array [
          "100%",
          "100%",
          "50%",
          "50%",
        ]
      }
    >
      <img
        className="c9 c10"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FwHFgQlrTrHav5O6bQRJ0dg%252FUntitled%2BSuspended_30x67x33%2B%25282%2529_sm%2Bcropped.jpg&width=680&height=450&quality=95"
      />
      <div
        className="c11"
      >
        <svg
          className="c12"
          version="1.1"
          viewBox="0 0 40 56"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <g
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <polygon
              fill="#000"
              points="0 0 0 56 39.3600006 28"
            />
          </g>
        </svg>
        <div
          className="c13"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize="14px"
        >
          02:28
        </div>
      </div>
    </div>
  </div>
</a>
`;
